src/aes/openssl/aes.c

```diff
@@ -128,6 +153,147 @@ int aes_encr(struct aes *aes, uint8_t *out, const uint8_t *in, size_t len)
 }
 
 
+/**
+ * Add (append) any AAD (Additional Authenticated Data) bytes.
+ *
+ * This can be called zero or more times as required
+ *
+ * @note Only valid for AEAD ciphers
+ *
+ * @param aes     AES context
+ * @param aad     Additional Authenticated Data (AAD)
+ * @param aad_len Number of AAD bytes
+ *
+ * @return 0 if success, otherwise errorcode
+ */
+int aes_add_aad(struct aes *aes, const uint8_t *aad, size_t aad_len)
+{
+	int tmplen;
+
+	if (!aes || !aad || !aad_len)
+		return EINVAL;
+
+	switch (aes->mode) {
+
+	case AES_MODE_GCM:
+		break;
+	default:
+		return ENOTSUP;
+	}
+
+	if (!EVP_CipherUpdate(aes->ctx, NULL, &tmplen, aad, (int)aad_len)) {
```

Comment:
`size_t` to `int` cast should probably be checked.

Reply:
do you mean that the casting is not needed ?

many openssl functions are using `int` type for counting bytes:

```
int EVP_CipherUpdate(EVP_CIPHER_CTX *ctx, unsigned char *out, int *outl,
                     const unsigned char *in, int inl);
```

Reply:
The casting is needed but I'd do checks such as `aad_len > INT_MAX`.